scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -376,6 +373,155 @@ class FormatWriter(formatOps: FormatOps) {
       }
 
     }
+
+    /**
+      * Returns how many extra spaces are needed to align tokens, as configured
+      * by `initStyle.align.tokens`.
+      */
+    // TODO(olafur) Refactor implementation to make it maintainable. It's super
+    // imperative and error-prone right now.
+    private def alignmentTokens: Map[TokenHash, Int] = {
+      lazy val noAlignTokens = initStyle.align.tokens.isEmpty &&
+        styleMap.tok2style.values.forall(_.align.tokens.isEmpty)
+      if (locations.length != tokens.length || noAlignTokens)
+        Map.empty[TokenHash, Int]
+      else {
+        var columnShift = 0
+        val finalResult = Map.newBuilder[TokenHash, Int]
+
+        // all blocks must be here, to get final flush
+        val blocks = new mutable.HashMap[Tree, AlignBlock]
+        def createBlock(x: Tree) = blocks.getOrElseUpdate(x, new AlignBlock)
+        var prevAlignContainer: Tree = null
+        var prevBlock: AlignBlock =
+          if (initStyle.align.multiline) null else createBlock(null)
```

Comment:
What is purpose of `createBlock(null)` here? Is it necessary to have map with single empty block on line 456?

Reply:
i considered that adding a map with a single entry for an entire document is very cheap. otherwise, see two comments below, that code would have to be modified to distinguish between multiline and non-multiline, which increases clutter.

but if you prefer, i can easily change that.

Reply:
No, it's just clarification 👍